tests/framework/utils.py

```diff
@@ -647,13 +647,38 @@ def generate_mmds_session_token(ssh_connection, ipv4_address, token_ttl):
     return token
 
 
-def generate_mmds_v2_get_request(ipv4_address, token, app_json=True):
+def generate_mmds_get_request(ipv4_address, token=None, app_json=True):
     """Build `GET` request to fetch metadata from MMDS when using V2."""
     cmd = 'curl -m 2 -s'
-    cmd += ' -X GET'
-    cmd += ' -H  "X-metadata-token: {}"'.format(token)
+
+    if token is not None:
+        cmd += ' -X GET'
+        cmd += ' -H  "X-metadata-token: {}"'.format(token)
+
     if app_json:
         cmd += ' -H "Accept: application/json"'
+
     cmd += ' http://{}/'.format(ipv4_address)
 
     return cmd
+
+
+def configure_mmds(test_microvm, iface_id, version, ipv4_address=None,
```

Comment:
Let's accept `[iface_id]` instead of `iface_id` as a parameter here to make it more flexible to future use.